sky/utils/controller_utils.py

```diff
@@ -717,26 +723,43 @@ def maybe_translate_local_file_mounts_and_sync_up(task: 'task_lib.Task',
             ux_utils.spinner_message(
                 f'Translating {msg} to SkyPilot Storage...'))
 
+    # Get the bucket name for the workdir and file mounts,
+    # we stores all these files in same bucket from config.
+    bucket_wth_prefix = skypilot_config.get_nested(('jobs', 'bucket'), None)
+    if bucket_wth_prefix is None:
+        store_type = sub_path = None
+        storage_account_name = region = None
+        allow_bucket_creation = True
+        bucket_name = constants.FILE_MOUNTS_BUCKET_NAME.format(
+            username=common_utils.get_cleaned_username(), id=run_id)
+    else:
+        store_type, bucket_name, sub_path, storage_account_name, region = \
+            storage_lib.StoreType.get_fields_from_store_url(bucket_wth_prefix)
+        allow_bucket_creation = False
+
     # Step 1: Translate the workdir to SkyPilot storage.
     new_storage_mounts = {}
     if task.workdir is not None:
-        bucket_name = constants.WORKDIR_BUCKET_NAME.format(
-            username=common_utils.get_cleaned_username(), id=run_id)
         workdir = task.workdir
         task.workdir = None
         if (constants.SKY_REMOTE_WORKDIR in original_file_mounts or
                 constants.SKY_REMOTE_WORKDIR in original_storage_mounts):
             raise ValueError(
                 f'Cannot mount {constants.SKY_REMOTE_WORKDIR} as both the '
                 'workdir and file_mounts contains it as the target.')
-        new_storage_mounts[
-            constants.
-            SKY_REMOTE_WORKDIR] = storage_lib.Storage.from_yaml_config({
-                'name': bucket_name,
-                'source': workdir,
-                'persistent': False,
-                'mode': 'COPY',
-            })
+        storage_obj = storage_lib.Storage(
+            name=bucket_name,
+            source=workdir,
+            persistent=False,
+            mode=storage_lib.StorageMode.COPY,
+            _bucket_sub_path=_sub_path_join(
+                sub_path,
+                constants.FILE_MOUNTS_WORKDIR_SUBPATH.format(run_id=run_id)))
+        if store_type is not None:
+            storage_obj.construct_store(store_type, region,
```

Comment:
See other comment on construct_store. Instead of this, can we initialize the store above and pass it through the stores arg `storage_lib.Storage(stores = ..., ...)`?